lib/utils.pm

```diff
@@ -1134,6 +1134,10 @@ sub ensure_serialdev_permissions {
     }
     else {
         assert_script_run "chown $testapi::username /dev/$testapi::serialdev && gpasswd -a $testapi::username \$(stat -c %G /dev/$testapi::serialdev)";
+        # when serial getty is started, it changes the group of serialdev from dialout to tty (but doesn't change it back when stopped)
+        # let's make sure that both will work
+        script_run "gpasswd -a $testapi::username tty";
+        script_run "gpasswd -a $testapi::username dialout";
```

Comment:
we already add the same user to one of the two groups in line L1136. I recommend to not do it again here, or rather replace the whole block by:

```
# when serial getty is started, it changes the group of serialdev from dialout to tty (but doesn't change it back when stopped)
# let's make sure that both will work
assert_script_run "chown $testapi::username /dev/$testapi::serialdev && usermod -a -G tty dialout $testapi::username \$(stat -c %G /dev/$testapi::serialdev)";
```

Reply:
This would lead to problems if for some reason the device would be owned by a group other than `tty` or `dialout`.
My solution will work in that case.

Reply:
done